packages/radix-ui-themes/src/helpers/props/color.prop.ts

```diff
@@ -2,6 +2,12 @@ import { themePropDefs } from '../../theme-options';
 import type { PropDef } from '..';
 
 const colorProp = {
```

Comment:
I think I'm directionally right about the outcome but not sure if there is a better way to implement. Essentially:
- Most components will use `colorProp` and get an empty `data-accent-color` attribute if you don't pass an explicit `color`. This defines the right high-contrast colors for Text, Heading, Link, etc. without having to whitelist which components have to do that. As the result, a couple of new component combinations would now work out of the box, e.g. like this:
  ```jsx
  <Badge color="gray">
    Status: <Text highContrast>Backlog</Text>
  </Badge>
  ```
- Components like Text, Heading, Link that need to be able to derive their color from a parent when their own color is not set explicitly would use `inheritedColorProp`